swift/ql/src/queries/Security/CWE-089/SqlInjection.qhelp

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+
+<p>
+If a database query (such as an SQL query) is built from user-provided data without sufficient sanitization, a user may be able to run malicious database queries.
+</p>
+
+</overview>
+<recommendation>
+
+<p>
+Most database connector libraries offer a way of safely embedding untrusted data into a query by means of query parameters or prepared statements. Use these features rather than building queries by string concatenation.
```

Comment:
In the context, is "user-provided data without sufficient sanitization" equivalent to "building queries by string concatenation"?

Reply:
Yes this is the same idea, though they're not quite formally equivalent.  Concatenation will be by far the most common case.  I'll think about whether we can be more consistent describing this.